### PR TITLE
Feature (frontend): Add filtering for server names

### PR DESCRIPTION
**Pull request by SmillaDion**
# Changes
* Added search bar
* Nodes change opacity depending on whether a search matched on a server name
* SVG is responsive so it resizes with the window
* SVG is placed behind HTML elements

# How was this tested?

* By refreshing 1000 times

# Type of change

Please check all boxes that apply to this change below.

| Type                     | Applies ✅|
| ------------------------ | :-------: |
| Backend                  |           |
| Frontend                 |      X     |
| Github/Ci/Admin          |           |
| Documentation required   |           |

# Checklist

Please follow the checklist below and check all completed items as you work on the PR.

I have..
- [X] performed a self-review of my own code
- [X] commented my code, particularly in hard-to-understand areas
- [ ] written documentation
- [ ] added automated tests
- [X] merged dependencies to `develop

### CODE REVIEW

File: src/components/Graph.vue [hunk 1]
```diff
@@ -1,14 +1,141 @@
...
+      }
+      return serverName.toLowerCase().includes(this.searchText.toLowerCase()) // Boolean statement
+    },
+    // Fetches data from backend and updates data variables
+    async getData () {
+      const varzResponse = await axios.get('https://localhost:5001/nodes')
+      this.servers = varzResponse.data
+
+      const routezResponse = await axios.get('https://localhost:5001/links')
+      this.routes = routezResponse.data
```
Comment (Maasdamind) [inline src/components/Graph.vue:65]:
```suggestion
    async getData () {
      const varzResponse = await axios.get('https://localhost:5001/nodes')
      this.servers = varzResponse.data

      const routezResponse = await axios.get('https://localhost:5001/links')
      this.routes = routezResponse.data
```

We shouldn't put localhost in there directly.  I'd suggest making environment variables called something like `APP_URL` and `APP_PORT`, and then fetching from there

**Comment by tjomson** [comment]
Super sleek looking design.
I would like the search field to also search for server-id, as i think that is a more common use-case than searching for the name.

I am wondering if we should add some secondary colors that we set the nodes to instead of making them translucent, such that you can't see the links under the nodes. Maybe this will be messy, and i guess it can be added later if we want to, so it is no biggie.
billede

I can see that the links also change their opacity depending on the search, but they are not part of the search result. Do we maybe want the links connected to a node in the search result to light up? Again, not really something we need to change, but i think it would be interesting to ask Andrii if that would be a good feature.

**Comment by SmillaDion** [comment]
> 
> 
> Super sleek looking design.
> I would like the search field to also search for server-id, as i think that is a more common use-case than searching for the name.
> 
> I am wondering if we should add some secondary colors that we set the nodes to instead of making them translucent, such that you can't see the links under the nodes. Maybe this will be messy, and i guess it can be added later if we want to, so it is no biggie.
> billede
> 
> I can see that the links also change their opacity depending on the search, but they are not part of the search result. Do we maybe want the links connected to a node in the search result to light up? Again, not really something we need to change, but i think it would be interesting to ask Andrii if that would be a good feature.

I think we should ask Andrii about whether the search bar should be for name or ID, or both.

Regarding translucency, I thought about using a secondary color for the nodes, but ultimately decided not to for now as we haven't chosen a color scheme or design for our graph yet. But I agree that once we have, that would be a good idea.

For the links I did attempt to implement what you suggest, but I couldn't get it working. For some reason the links just disappear when I do it. I thought it may be fine either way to only show nodes, because we are only searching for node identifiers. But if we can figure out how to do it, and if Andrii likes it, then sure thing.

**Comment by hojelse** [comment]
> # How was this tested?
> By refreshing 1000 times
👌

**Review by tjomson — APPROVED** [review]
Ye looks nice. We can add some additional cool stuff later if we have time for it.

**Pull request merged**